File: nodejs/signature.js
const crypto = require('crypto');

const query_string = 'timestamp=1578963600000';
const apiSecret = 'NhqPtmdSJYdKjVHjA7PZj4Mge3R5YNiP1e3UZjInClVN65XAbvqqM6A7H5fATj0j';

function signature(query_string) {
    return crypto
        .createHmac('sha256', apiSecret)
        .update(query_string)
        .digest('hex');
}

console.log("hashing the string: ");
console.log(query_string);
console.log("and return:");
console.log(signature(query_string));

console.log("\n");

const another_query = 'symbol=LTCBTC&side=BUY&type=LIMIT&timeInForce=GTC&quantity=1&price=0.1&recvWindow=5000&timestamp=1499827319559';
console.log("hashing the string: ");
console.log(another_query);
console.log("and return:");
console.log(signature(another_query));

File: perl/signature.pl
#!/usr/bin/perl
use v5.010;
use warnings;
use strict;
use Digest::SHA qw(hmac_sha256_hex); 
my $secretKey = "NhqPtmdSJYdKjVHjA7PZj4Mge3R5YNiP1e3UZjInClVN65XAbvqqM6A7H5fATj0j";
my $exampleQuery1 = "timestamp=1578963600000";
say "example query string 1: ", $exampleQuery1;
say hmac_sha256_hex($exampleQuery1, $secretKey);

my $exampleQuery2 = "symbol=LTCBTC&side=BUY&type=LIMIT&timeInForce=GTC&quantity=1&price=0.1&recvWindow=5000&timestamp=1499827319559";
say "example query string 2: ", $exampleQuery2;
say hmac_sha256_hex($exampleQuery2, $secretKey);
